plugins/src/gcc/imp.rs

```diff
@@ -0,0 +1,1274 @@
+    packets: Vec<Packet>,
+    departure: Duration,       // ms
+    arrival: Option<Duration>, // ms
+}
+
+impl Default for PacketGroup {
+    fn default() -> Self {
+        Self {
+            packets: Default::default(),
+            departure: Duration::zero(),
+            arrival: None,
+        }
+    }
+}
+
+fn pdur(d: &Duration) -> String {
+    let stdd = time::Duration::from_nanos(d.num_nanoseconds().unwrap().abs() as u64);
+
+    format!("{}{stdd:?}", if d.lt(&Duration::zero()) { "-" } else { "" })
+}
+
+impl PacketGroup {
+    fn add(&mut self, packet: Packet) {
+        if self.departure.is_zero() {
+            self.departure = packet.departure;
+        }
+
+        self.arrival = Some(
+            self.arrival
+                .map_or_else(|| packet.arrival, |v| v.max(packet.arrival)),
+        );
+
+        self.packets.push(packet);
+    }
+
+    /// Returns the delta between self.arrival_time and @prev_group.arrival_time in ms
+    // t(i) - t(i-1)
+    fn inter_arrival_time(&self, prev_group: &Self) -> Duration {
+        // Should never be called if we haven't gotten feedback for all
+        // contained packets
+        self.arrival.unwrap() - prev_group.arrival.unwrap()
+    }
+
+    fn inter_arrival_time_pkt(&self, next_pkt: &Packet) -> Duration {
+        // Should never be called if we haven't gotten feedback for all
+        // contained packets
+        next_pkt.arrival - self.arrival.unwrap()
+    }
+
+    /// Returns the delta between self.departure_time and @prev_group.departure_time in ms
+    // T(i) - T(i-1)
+    fn inter_departure_time(&self, prev_group: &Self) -> Duration {
+        // Should never be called if we haven't gotten feedback for all
+        // contained packets
+        self.departure - prev_group.departure
+    }
+
+    fn inter_departure_time_pkt(&self, next_pkt: &Packet) -> Duration {
+        // Should never be called if we haven't gotten feedback for all
+        // contained packets
+        next_pkt.departure - self.departure
+    }
+
+    /// Returns the delta between intern arrival time and inter departure time in ms
+    fn inter_delay_variation(&self, prev_group: &Self) -> Duration {
+        // Should never be called if we haven't gotten feedback for all
+        // contained packets
+        self.inter_arrival_time(prev_group) - self.inter_departure_time(prev_group)
+    }
+
+    fn inter_delay_variation_pkt(&self, next_pkt: &Packet) -> Duration {
+        // Should never be called if we haven't gotten feedback for all
+        // contained packets
+        self.inter_arrival_time_pkt(next_pkt) - self.inter_departure_time_pkt(next_pkt)
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+enum NetworkUsage {
+    Normal,
+    Over,
+    Under,
+}
+
+struct Detector {
+    // Grouping fields
+    group: PacketGroup,
+    prev_group: Option<PacketGroup>,
+    measure: Duration, // Delay variation measure
+
+    // FIXME: This is not used but should be according 5.3:
```

Comment:
Let's try to re-enable this now that you have something that works :)

Reply:
Removed it in the end and added a comment explainning.